CONTRIBUTING.md

```diff
@@ -2,133 +2,116 @@
+
+### Integration Tests
+
+We create various miminal dash apps to cover feature scenario. A server is launched in mutli-thread or multi-process flavor and the test steps are executed in browsers driving by selenium webdrivers.
+
+Any reasonable test scenario is encouraged to be added along with the same PR.
+
+### Visual regression with Percy
+
+Testing graph-intensive-application is a challenging job. We use [percy](https://percy.io/) to mitigate the pain, please pay attention if percy reports visual differences. If you are not sure whether the change is expected, leave a comment, and don't blind-approve it.
+
+
+## Test variable tips
 You can configure the test server with the following variables:
+
 ### DASH_TEST_CHROMEPATH
 If you run a special chrome, set the path to your chrome binary with this environment variable.
 
 ### DASH_TEST_PROCESSES
 If you encounter errors about Multi-server + Multi-processing when running under Python 3, try running the tests with the number of server processes set to 1.
 
 ### Example: single test run with configuration
-```
-DASH_TEST_CHROMEPATH=/bin/google-chrome-beta DASH_TEST_PROCESSES=1 python -m unittest -v tests.test_integration.Tests.test_no_callback_context
-```
 
-## Making a contribution
-_For larger features, your contribution will have a higher likelihood of getting merged if you create an issue to discuss the changes that you'd like to make before you create a pull request._
-
-1. Create a pull request and tag the Plotly team (`@plotly/dash`) and tag.
-2. After a review has been done and your changes have been approved, create a prerelease and comment in the PR. Version numbers should follow [semantic versioning][]. To create a prerelease:
-    * Add `rc1` to `version.py` (`./dash/version.py`) e.g. `0.13.0rc1`
-        - If needed, ask @chriddyp to get PyPi package publishing access.
-    * Run `python setup.py sdist` to build a distribution zip.
-    * Check the `dist` folder for a `tar.gz` file ending with your selected version number. Double check that this version number ends with `rc#`, as to not mistakenly publish the package.
-    * Run `twine upload dist/<package_name>`.
-3. Comment in the PR with the prerelease version
-4. Update the top-level comment to include info about how to install, a summary of the changes, and a simple example.
-    * This makes it easier for a community member to come in and try it out. As more folks review, it's harder to find the installation instructions deep in the PR
-    * Keep this top-level comment updated with installation instructions (e.g. the `pip install` command)
-5. Make a post in the [Dash Community Forum][]
-    * Title it `":mega: Announcement! New <Your Feature> - Feedback Welcome"`
-    * In the description, link to the PR and any relevant issue(s)
-    * Pin the topic so that it appears at the top of the forum for two weeks
-
-## [Checklists](http://rs.io/unreasonable-effectiveness-of-checklists/)
```

Comment:
@alexcjohnson  I will move the checklist section into a template file https://help.github.com/en/articles/creating-a-pull-request-template-for-your-repository